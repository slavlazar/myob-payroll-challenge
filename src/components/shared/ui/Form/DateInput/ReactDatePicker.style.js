import { css } from 'styled-components';

const ReactDatePicker = css`
  .react-datepicker-wrapper,
  .react-datepicker__input-container {
    display: block;
  }
  .react-datepicker {
    font-size: 1.2rem;
    font-family: ${props => props.theme.font.family.heading};
    font-weight: ${props => props.theme.font.weight.bold};
    line-height: ${props => props.theme.lineHeight.text};
  }
  .react-datepicker__current-month,
  .react-datepicker-time__header,
  .react-datepicker-year-header {
    font-size: 1.3rem;
  }
  .react-datepicker__month--selected,
  .react-datepicker__month--in-selecting-range,
  .react-datepicker__month--in-range {
    background-color: ${props => props.theme.colour.primary};
  }
`;

export default ReactDatePicker;
