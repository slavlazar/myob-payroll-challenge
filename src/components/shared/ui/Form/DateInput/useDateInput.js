import moment from 'moment';

/**
 * Hook to handle field touch and change values if values are valid dates
 * @param {function} setFieldValue Formik function to update field value
 * @param {function} setFieldTouched Formik function to set field touched
 * @param {string} name Field name
 */
export const useDateInput = (setFieldValue, setFieldTouched, name) => {
  const handleChangeRaw = e => {
    const { name: targetName, value } = e.target;
    const validChars = /^\d{0,2}[./]{0,1}\d{0,2}[./]{0,1}\d{0,4}$/;
    if (!validChars.test(value)) {
      e.preventDefault();
    }

    if (moment(new Date(value)).isValid()) {
      setFieldValue(targetName, value);
      setFieldTouched(targetName, true);
    }
  };

  const handleChange = date => {
    const dateInstance = new Date(date);
    if (moment(dateInstance).isValid()) {
      setFieldValue(name, dateInstance);
      setFieldTouched(name, true);
    }
  };

  return [handleChangeRaw, handleChange];
};
