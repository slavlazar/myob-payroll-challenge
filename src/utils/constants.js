export const LOGIN_URL = '/login';
export const API_LOGIN_ENDPOINT = '';
