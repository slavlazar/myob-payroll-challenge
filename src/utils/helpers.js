/**
 * Helper method to get file extension of given file name
 * @param {string} filename Full filename
 * @return {string} Given filename, will return the extension
 */
export const getFileExtension = filename =>
  filename
    .split('.')
    .pop()
    .replace(/[\W_]+/g, '');

/**
 * Helper method to general initial values from a given schema
 * @param {array} fields Array of fields to generate initial values
 * @return {object} Object of generated initial values from schema
 */
export const generateInitialValuesFromSchema = (fields = []) =>
  fields.reduce((accum, key) => ({ ...accum, [key]: '' }), {});

/**
 * Helper method to determine if given object is empty
 * @param {object} obj Object to check if empty
 * @return {boolean} Given object, return true if empty, false otherwise
 */
export const isObjectEmpty = obj =>
  Object.keys(obj).length === 0 && obj.constructor === Object;

/**
 * Helper method to log error messages to console
 * Curried function that first takes name of group,
 * followed by console messages
 * @param {string} groupName Name of group to log
 * @returns {[function, void]} Either returns a curried function or logs
 * to console
 */
export const logError = groupName => (...consoleMsg) => {
  console.group(groupName);
  consoleMsg.forEach(msg => {
    console.error(msg);
  });
  console.groupEnd();
};

/**
 * Helper method to round number to nearest integer
 * @param {number} val Number to round
 * @return {number} Rounded number
 */
export const round = val => Math.round(val);

/**
 *
 * @param {number} val Number to format
 * @param {boolean} roundValue Whether to round the value before formatting
 * @return {string} Formatted currency value
 */
export const formatCurrency = (val, roundValue = true) => {
  const valToConvert = roundValue ? round(val) : val;

  return Number(valToConvert).toLocaleString('en-AU', {
    style: 'currency',
    currency: 'AUD',
    minimumFractionDigits: 2,
  });
};
