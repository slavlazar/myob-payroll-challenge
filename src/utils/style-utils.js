import React from 'react';
import { mount } from 'enzyme';
import { css, ThemeProvider } from 'styled-components';

import theme from '../theme';

const { media: sizes } = theme;

// Media query, usage: ${media.tablet`padding: 1.5rem`}
// min-width used for mobile-first design
export const media = Object.keys(sizes).reduce((acc, label) => {
  acc[label] = (...args) => css`
    @media (min-width: ${sizes[label] / 16}em) {
      ${css(...args)}
    }
  `;

  return acc;
}, {});

export const renderWithTheme = component => {
  return mount(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};
