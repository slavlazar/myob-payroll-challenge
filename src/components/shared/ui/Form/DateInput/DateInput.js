import React from 'react';
import PropTypes from 'prop-types';
import DatePicker from 'react-datepicker';
// Require datepicker styles for general date picker display
// We will modify this with styled components
import 'react-datepicker/dist/react-datepicker.css';

import Input from '../Input/Input.style';
import Field from '../Field';
import { useDateInput } from './useDateInput';

/**
 * Displays React Datepicker object styled with our Field base
 * component styles
 * @param {object} props DatePicker props to generate
 */
const DateInput = props => {
  const {
    formik: { setFieldValue, setFieldTouched },
    dateFormat,
    disabled,
    name,
    label,
    id,
    placeholder,
    onFocus,
    onBlur,
    showMonthYearPicker,
    className,
    errorMessage,
    ...rest
  } = props;

  const [handleChangeRaw, handleChange] = useDateInput(
    setFieldValue,
    setFieldTouched,
    name
  );

  const { value } = rest;

  return (
    <Field
      label={label}
      errorMessage={errorMessage}
      value={value}
      disabledKeyboardNavigation
      autoComplete="off"
      {...rest}
      name={name}
      dateFormat={dateFormat}
      selected={value ? new Date(value) : null}
      error={errorMessage}
      placeholderText={placeholder}
      showMonthYearPicker={showMonthYearPicker}
      onChangeRaw={handleChangeRaw}
      onChange={handleChange}
      onFocus={onFocus}
      onBlur={onBlur}
      disabled={disabled}
      render={renderProps => (
        <Input as={DatePicker} className={className} {...renderProps} />
      )}
    />
  );
};

DateInput.propTypes = {
  /** Form values from Formik */
  formik: PropTypes.instanceOf(Object).isRequired,
  /** Function to handle field blur events */
  onBlur: PropTypes.func.isRequired,
  /** Name of input field */
  name: PropTypes.string.isRequired,
  onFocus: PropTypes.func,
  id: PropTypes.string,
  className: PropTypes.string,
  style: PropTypes.instanceOf(Object),
  label: PropTypes.string,
  hint: PropTypes.string,
  dateFormat: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]),
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  disabled: PropTypes.bool,
  showMonthYearPicker: PropTypes.bool,
  errorMessage: PropTypes.string,
};

DateInput.defaultProps = {
  onFocus: () => {},
  id: null,
  className: null,
  style: null,
  label: null,
  hint: null,
  dateFormat: 'MM/yyyy',
  placeholder: null,
  required: false,
  disabled: false,
  showMonthYearPicker: true,
  errorMessage: null,
};

export default DateInput;
