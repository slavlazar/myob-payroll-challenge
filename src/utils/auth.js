import axios from 'axios';

import { API_LOGIN_ENDPOINT } from './constants';

/**
 * Login method that calls upon our API endpoint to log the user in
 * @param {object} Object of email and password keys to log user in with
 * @async
 */
const login = async ({ email, password }) => {
  const { data: userInfo } = await axios.post(API_LOGIN_ENDPOINT, {
    email,
    password,
  });

  return userInfo;
};

export default {
  login,
};
