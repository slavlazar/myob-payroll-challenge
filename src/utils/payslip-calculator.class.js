import moment from 'moment';

import * as TAXABLE_INCOME from '../config/tax-rates.json';

import { logError as logErrorGroup, formatCurrency } from './helpers';

/** Class representing Payslip calculator. */
class PayslipCalculator {
  static logError = logErrorGroup('PayslipCalculator');

  /**
   * Create a payslip calculator.
   * @param {string} firstName - Employee first name.
   * @param {string} lastName - Employee last name.
   * @param {number} annualSalary - Employee annual salary.
   * @param {number} superRate - Employee super rate.
   */
  constructor(employeeDetails) {
    ({
      firstName: this._firstName,
      lastName: this._lastName,
      annualSalary: this._annualSalary,
      superRate: this._superRate,
    } = employeeDetails);

    this._grossIncome = 0;
    this._incomeTax = 0;
    this._netIncome = 0;
    this._superTotal = 0;
    this._payFrequency = 'Monthly';

    this._determinePayPeriod(employeeDetails.payPeriod);
  }

  /**
   * Get employee full name.
   * @return {string} Employee first name and last name.
   */
  get name() {
    return `${this._firstName} ${this._lastName}`;
  }

  /**
   * Get employee pay period.
   * @return {string} Employee pay period based on month and year.
   */
  get payPeriod() {
    return this._payPeriod;
  }

  /**
   * Get employee pay frequency.
   * @return {string} Employee pay frequency - Defaults to monthly for demo.
   */
  get payFrequency() {
    return this._payFrequency;
  }

  /**
   * Get employee annual salary.
   * @return {string} Employee salary formatted to AUD currency.
   */
  get annualSalary() {
    return formatCurrency(this._annualSalary);
  }

  /**
   * Get employee gross income.
   * @return {string} Employee gross income formatted to AUD currency.
   */
  get grossIncome() {
    return formatCurrency(this._grossIncome);
  }

  /**
   * Get employee income tax.
   * @return {string} Employee income tax formatted to AUD currency.
   */
  get incomeTax() {
    return formatCurrency(this._incomeTax);
  }

  /**
   * Get employee net income.
   * @return {string} Employee net income formatted to AUD currency.
   */
  get netIncome() {
    return formatCurrency(this._netIncome);
  }

  /**
   * Get employee super total.
   * @return {string} Employee super total formatted to AUD currency.
   */
  get superTotal() {
    const superCalc = this._superRate / 100;
    return formatCurrency(this._grossIncome * superCalc);
  }

  /**
   * Get employee pay.
   * @return {string} Employee pay for the month, formatted to AUD currency.
   */
  get pay() {
    return formatCurrency(this._netIncome - this._superTotal);
  }

  /**
   * Get total tax amount for employee, given taxable amount,
   * tax rate and base tax amount.
   * @static
   * @param {number} Taxable amount of employee
   * @returns [{function}, {function}, {number}] Total tax amount for employee
   */
  static __totalTaxAmount(taxable) {
    return taxRate => baseTaxAmount => baseTaxAmount + taxable * taxRate;
  }

  /**
   * Determine pay period (currently monthly) of given date
   * @param {date} payPeriod Date object of current pay period
   * @return {string} Formatted string of pay period from start of
   * month to end of month
   */
  _determinePayPeriod(payPeriod) {
    const payPeriodMoment = moment(payPeriod);
    if (payPeriodMoment.isValid()) {
      const dateFormat = 'DD MMMM';
      const endPayPeriodMoment = payPeriodMoment.clone();
      payPeriodMoment.startOf('month');
      endPayPeriodMoment.endOf('month');

      this._payPeriod = `${payPeriodMoment.format(
        dateFormat
      )} to ${endPayPeriodMoment.format(dateFormat)}`;
    }
  }

  /**
   * Calculate gross based on annual salary / 12 months
   * @return {number} Gross income calculated
   */
  _calculateGross() {
    let grossIncome = 0;

    try {
      grossIncome = this._annualSalary / 12;
    } catch (e) {
      PayslipCalculator.logError(
        '_calculateGross',
        `Salary: ${this._annualSalary}`,
        e
      );
    }

    return grossIncome;
  }

  /**
   * Calculate income tax based on tax limit
   * Method will first reverse the provided TAXABLE_INCOME
   * values (so we can traverse in reverse order), then map over
   * this reversed array to find the correct taxable income based
   * on the employees annual salary
   * @return {number} Income tax based on employee salary and taxable
   * income rules
   */
  _calculateIncomeTax() {
    let totalTax = 0;

    // Avoid mutation/side effect
    const reversedTaxRates = [...TAXABLE_INCOME.default].reverse();

    reversedTaxRates.some(tax => {
      if (this._annualSalary > tax.limit) {
        try {
          const taxable = PayslipCalculator.__totalTaxAmount(
            this._annualSalary - tax.limit
          );
          totalTax = taxable(tax.taxRate)(tax.baseTaxAmount);
        } catch (e) {
          PayslipCalculator.logError(
            'Could not calculate income tax',
            `this._annualSalary: ${this._annualSalary}`,
            `tax object: ${tax}`,
            e
          );
        }
        return true;
      }

      return false;
    });

    return totalTax > 0 ? totalTax / 12 : 0;
  }

  /**
   * Calculates net income based on gross income and income tax
   * @return {number} Net income total
   */
  _calculateNetIncome() {
    return this._grossIncome - this._incomeTax;
  }

  /**
   * Calculates Super based on gross income and super rate
   * @return {number} Super value
   */
  _calculateSuperTotal() {
    const superCalc = this._superRate / 100;
    return this._grossIncome * superCalc;
  }

  /**
   * Public method to generate payslip based on provided variables
   * @return {void}
   * @public
   */
  generate() {
    this._grossIncome = this._calculateGross();
    this._incomeTax = this._calculateIncomeTax();

    this._netIncome = this._calculateNetIncome();
    this._superTotal = this._calculateSuperTotal();
  }
}

export default PayslipCalculator;
