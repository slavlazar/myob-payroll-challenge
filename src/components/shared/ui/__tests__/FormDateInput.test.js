import React from 'react';
import sinon from 'sinon';
import DatePicker from 'react-datepicker';

import DateInput from '../Form/DateInput/DateInput';
import { renderWithTheme } from '../../../../utils/style-utils';

const baseProps = {
  name: 'DateInput',
  onFocus: sinon.spy(),
  onBlur: sinon.spy(),
  value: 'Testing',
  formik: {
    setFieldValue: sinon.spy(),
    setFieldTouched: sinon.spy(),
  },
};

const mountWrapper = (props = {}) => {
  const allProps = {
    ...baseProps,
    ...props,
  };

  return renderWithTheme(<DateInput {...allProps} />);
};

describe('Components: Shared: UI: Form DateInput', () => {
  it('renders correctly', () => {
    const wrapped = mountWrapper();
    expect(wrapped).toMatchSnapshot();
  });

  it('should render a DatePicker', () => {
    const wrapped = mountWrapper();
    expect(wrapped.find(DatePicker)).toHaveLength(1);
  });

  it('should be disabled', () => {
    const wrapper = mountWrapper({ disabled: true });

    expect(wrapper.find('input').prop('disabled')).toBe(true);
  });
});
