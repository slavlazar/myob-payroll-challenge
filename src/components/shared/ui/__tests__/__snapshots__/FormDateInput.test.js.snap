// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Shared: UI: Form DateInput renders correctly 1`] = `
.c2 {
  width: 100%;
  border: none;
  box-sizing: border-box;
  outline: none;
  font-size: 1.6rem;
  padding: 1rem 1.6rem;
  border-radius: 4px;
  background-color: #ffffff;
  display: block;
  border-width: 2px;
  border-style: solid;
  border-color: #efefef;
}

.c2:focus {
  outline: none;
  -webkit-box-shadow: 0 0.5rem 2rem rgba(0,0,0,0.1);
  -moz-box-shadow: 0 0.5rem 2rem rgba(0,0,0,0.1);
  box-shadow: 0 0.5rem 2rem rgba(0,0,0,0.1);
  border-color: #8241AA;
}

.c0 {
  margin-bottom: 1.5rem;
}

.c1 {
  margin: 0;
}

.c1 .react-datepicker-wrapper,
.c1 .react-datepicker__input-container {
  display: block;
}

.c1 .react-datepicker {
  font-size: 1.2rem;
  font-family: NeutroMYOB-Medium;
  font-weight: 600;
  line-height: 1.7;
}

.c1 .react-datepicker__current-month,
.c1 .react-datepicker-time__header,
.c1 .react-datepicker-year-header {
  font-size: 1.3rem;
}

.c1 .react-datepicker__month--selected,
.c1 .react-datepicker__month--in-selecting-range,
.c1 .react-datepicker__month--in-range {
  background-color: #8241AA;
}

<ThemeProvider
  theme={
    Object {
      "colour": Object {
        "alert": Object {
          "danger": "#fff5f7",
          "default": "#f5f5f5",
          "success": "#00d1b2",
        },
        "background": "#f3f4f5",
        "border": "#efefef",
        "buttonPrimary": "#ffffff",
        "buttonPrimaryBackground": "#00aa65",
        "footerBackground": "#1b2830",
        "heading": "#2e3e4f",
        "navBackground": "#ffffff",
        "primary": "#8241AA",
        "secondary": "#ed005f",
        "text": "#515e6d",
      },
      "font": Object {
        "family": Object {
          "body": "NeutroMYOB-Regular",
          "fallback": "-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,sans-serif",
          "heading": "NeutroMYOB-Medium",
          "strong": "NeutroMYOB-Demi",
        },
        "size": Object {
          "large": "1.8rem",
          "medium": "1.6rem",
          "small": "1.4rem",
          "xlarge": "2.2rem",
          "xsmall": "1.2rem",
        },
        "variants": Object {
          "body": Object {
            "family": "NeutroMYOB-Regular",
            "src": Array [
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Regular.woff2)",
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Regular.woff)",
            ],
            "style": "normal",
            "weight": 400,
          },
          "heading": Object {
            "family": "NeutroMYOB-Medium",
            "src": Array [
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Medium.woff2)",
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Medium.woff)",
            ],
            "style": "normal",
            "weight": 600,
          },
          "strong": Object {
            "family": "NeutroMYOB-Demi",
            "src": Array [
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Demi.woff2)",
              "url(https://assets.digital.myob.com/fonts/NeutroMYOB-Demi.woff)",
            ],
            "style": "normal",
            "weight": 700,
          },
        },
        "weight": Object {
          "bold": 600,
          "normal": 400,
        },
      },
      "lineHeight": Object {
        "heading": 1.5,
        "text": 1.7,
      },
      "media": Object {
        "desktop": 768,
        "large": 992,
        "tablet": 576,
      },
    }
  }
>
  <DateInput
    className={null}
    dateFormat="MM/yyyy"
    disabled={false}
    errorMessage={null}
    formik={
      Object {
        "setFieldTouched": [Function],
        "setFieldValue": [Function],
      }
    }
    hint={null}
    id={null}
    label={null}
    name="DateInput"
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder={null}
    required={false}
    showMonthYearPicker={true}
    style={null}
    value="Testing"
  >
    <InputComponent
      autoComplete="off"
      component={null}
      dateFormat="MM/yyyy"
      disabled={false}
      disabledKeyboardNavigation={true}
      error={null}
      errorMessage={null}
      hint={null}
      id={null}
      label={null}
      name="DateInput"
      onBlur={[Function]}
      onChange={[Function]}
      onChangeRaw={[Function]}
      onFocus={[Function]}
      placeholderText={null}
      render={[Function]}
      required={false}
      selected={Date { NaN }}
      showMonthYearPicker={true}
      style={null}
      value="Testing"
    >
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "lastClassName": "c0",
                "rules": Array [
                  "
  margin-bottom: 1.5rem;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c0"
          >
            <styled.div>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-EHOje",
                      "isStatic": false,
                      "lastClassName": "c1",
                      "rules": Array [
                        "
  margin: 0;
  ",
                        "
  .react-datepicker-wrapper,
  .react-datepicker__input-container {
    display: block;
  }
  .react-datepicker {
    font-size: 1.2rem;
    font-family: ",
                        [Function],
                        ";
    font-weight: ",
                        [Function],
                        ";
    line-height: ",
                        [Function],
                        ";
  }
  .react-datepicker__current-month,
  .react-datepicker-time__header,
  .react-datepicker-year-header {
    font-size: 1.3rem;
  }
  .react-datepicker__month--selected,
  .react-datepicker__month--in-selecting-range,
  .react-datepicker__month--in-range {
    background-color: ",
                        [Function],
                        ";
  }
",
                        ";
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-EHOje",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="c1"
                >
                  <styled.input
                    aria-describedby="DateInput_error"
                    as={[Function]}
                    autoComplete="off"
                    className={null}
                    dateFormat="MM/yyyy"
                    disabled={false}
                    disabledKeyboardNavigation={true}
                    error={null}
                    errorMessage={null}
                    hint={null}
                    id="DateInput"
                    name="DateInput"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onChangeRaw={[Function]}
                    onFocus={[Function]}
                    placeholderText={null}
                    required={false}
                    selected={Date { NaN }}
                    showMonthYearPicker={true}
                    style={null}
                    value="Testing"
                  >
                    <StyledComponent
                      aria-describedby="DateInput_error"
                      as={[Function]}
                      autoComplete="off"
                      className={null}
                      dateFormat="MM/yyyy"
                      disabled={false}
                      disabledKeyboardNavigation={true}
                      error={null}
                      errorMessage={null}
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bdVaJa",
                            "isStatic": false,
                            "lastClassName": "c2",
                            "rules": Array [
                              "
  width: 100%;
  border: none;
  box-sizing: border-box;
  outline: none;
  font-size: ",
                              [Function],
                              ";
  padding: 1rem 1.6rem;
  border-radius: 4px;
  background-color: ",
                              [Function],
                              ";
  display: block;
  border-width: 2px;
  border-style: solid;
  border-color: ",
                              [Function],
                              ";

  &:focus {
    outline: none;
    -webkit-box-shadow: 0 0.5rem 2rem rgba(0, 0, 0, 0.1);
    -moz-box-shadow: 0 0.5rem 2rem rgba(0, 0, 0, 0.1);
    box-shadow: 0 0.5rem 2rem rgba(0, 0, 0, 0.1);
    border-color: ",
                              [Function],
                              ";
  }
",
                            ],
                          },
                          "displayName": "styled.input",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bdVaJa",
                          "target": "input",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      hint={null}
                      id="DateInput"
                      name="DateInput"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onChangeRaw={[Function]}
                      onFocus={[Function]}
                      placeholderText={null}
                      required={false}
                      selected={Date { NaN }}
                      showMonthYearPicker={true}
                      style={null}
                      value="Testing"
                    >
                      <DatePicker
                        allowSameDay={false}
                        aria-describedby="DateInput_error"
                        autoComplete="off"
                        className="c2"
                        dateFormat="MM/yyyy"
                        dateFormatCalendar="LLLL yyyy"
                        disabled={false}
                        disabledKeyboardNavigation={true}
                        dropdownMode="scroll"
                        error={null}
                        errorMessage={null}
                        hint={null}
                        id="DateInput"
                        inlineFocusSelectedMonth={false}
                        monthsShown={1}
                        name="DateInput"
                        nextMonthButtonLabel="Next month"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onChangeRaw={[Function]}
                        onClickOutside={[Function]}
                        onFocus={[Function]}
                        onInputClick={[Function]}
                        onInputError={[Function]}
                        onKeyDown={[Function]}
                        onMonthChange={[Function]}
                        onSelect={[Function]}
                        onYearChange={[Function]}
                        placeholderText={null}
                        preventOpenOnFocus={false}
                        previousMonthButtonLabel="Previous Month"
                        readOnly={false}
                        renderDayContents={[Function]}
                        required={false}
                        selected={Date { NaN }}
                        shouldCloseOnSelect={true}
                        showMonthYearPicker={true}
                        showTimeInput={false}
                        showTimeSelect={false}
                        strictParsing={false}
                        style={null}
                        timeCaption="Time"
                        timeInputLabel="Time"
                        timeIntervals={30}
                        value="Testing"
                        withPortal={false}
                      >
                        <PopperComponent
                          hidePopper={true}
                          popperComponent={null}
                          popperModifiers={
                            Object {
                              "preventOverflow": Object {
                                "boundariesElement": "viewport",
                                "enabled": true,
                                "escapeWithReference": true,
                              },
                            }
                          }
                          popperPlacement="bottom-start"
                          popperProps={Object {}}
                          targetComponent={
                            <div
                              className="react-datepicker__input-container"
                            >
                              <input
                                autoComplete="off"
                                className="c2"
                                disabled={false}
                                id="DateInput"
                                name="DateInput"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                placeholder={null}
                                readOnly={false}
                                required={false}
                                type="text"
                                value="Testing"
                              />
                            </div>
                          }
                        >
                          <Manager>
                            <Reference>
                              <InnerReference
                                setReferenceNode={[Function]}
                              >
                                <div
                                  className="react-datepicker-wrapper"
                                >
                                  <div
                                    className="react-datepicker__input-container"
                                  >
                                    <input
                                      autoComplete="off"
                                      className="c2"
                                      disabled={false}
                                      id="DateInput"
                                      name="DateInput"
                                      onBlur={[Function]}
                                      onChange={[Function]}
                                      onClick={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      placeholder={null}
                                      readOnly={false}
                                      required={false}
                                      type="text"
                                      value="Testing"
                                    />
                                  </div>
                                </div>
                              </InnerReference>
                            </Reference>
                          </Manager>
                        </PopperComponent>
                      </DatePicker>
                    </StyledComponent>
                  </styled.input>
                </div>
              </StyledComponent>
            </styled.div>
          </div>
        </StyledComponent>
      </styled.div>
    </InputComponent>
  </DateInput>
</ThemeProvider>
`;
